src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 1) {
+      return this.toString();
+    }
+
+    let sep = separator;
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const sepCurrent = i === this.length - 1 ? '' : sep;
+
+      if (this[i] === null || this[i] === undefined) {
+        str += '' + sepCurrent;
+      } else {
+        str += this[i] + sepCurrent;
+      }
```

Comment:
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
        str += this[i];
      }
      
      if (i !== this.length - 1) {
        str += sepCurrent;
      }
```

Reply:
pros of this solution are: we didn't add empty string, if it's the last element, or if the current element is null or undefined